jobs/integration/validation.py

```diff
@@ -754,11 +754,15 @@ async def test_gpu_support(model, tools):
     """Test gpu support. Should be disabled if hardware
     is not detected and functional if hardware is fine"""
 
-    # See if the workers have nvidia
+    # See if any the workers have nvidia
     workers = model.applications["kubernetes-worker"]
-    action = await juju_run(workers.units[0], "lspci -nnk")
-    nvidia = True if action.stdout.lower().count("nvidia") > 0 else False
-
+    kubernetes_worker = workers.units[0]
+    for worker in workers.units:
+        action = await juju_run(worker, "lspci -nnk")
+        nvidia = True if action.stdout.lower().count("nvidia") > 0 else False
+        if nvidia:
+            kubernetes_worker = worker
+            break
     control_plane_unit = model.applications["kubernetes-control-plane"].units[0]
```

Comment:
Then setup a `verify_argfs` which can be used in subsequent calls to `verify_deleted` and `verify_ready`
```suggestion
    control_plane_unit = model.applications["kubernetes-control-plane"].units[0]
    verify_args = (
        control_plane,
        "ds",
        ["nvidia-device-plugin-daemonset"],
        "-n kube-system",
```